modules/planning/on_lane_planning.cc

```diff
@@ -613,110 +613,112 @@ void OnLanePlanning::ExportOpenSpaceChart(
+                       open_space_debug.xy_boundary(2) - 1.0,
+                       open_space_debug.xy_boundary(3) + 1.0, "y (meter)", true,
+                       chart);
 
-  if (open_space_debug.xy_boundary_size() == 4) {
-    PopulateChartOptions(open_space_debug.xy_boundary(0) - 1.0,
-                         open_space_debug.xy_boundary(1) + 1.0, "x (meter)",
-                         open_space_debug.xy_boundary(2) - 1.0,
-                         open_space_debug.xy_boundary(3) + 1.0, "y (meter)",
-                         true, chart);
-  }
-
-  if (open_space_debug.obstacles_size() != 0) {
-    int obstacle_index = 1;
-    for (const auto& obstacle : open_space_debug.obstacles()) {
-      auto* obstacle_outline = chart->add_line();
-      obstacle_outline->set_label("Bdr" + std::to_string(obstacle_index));
-      obstacle_index += 1;
-      for (int vertice_index = 0;
-           vertice_index < obstacle.vertices_x_coords_size(); vertice_index++) {
-        auto* point_debug = obstacle_outline->add_point();
-        point_debug->set_x(obstacle.vertices_x_coords(vertice_index));
-        point_debug->set_y(obstacle.vertices_y_coords(vertice_index));
-      }
-      // Set chartJS's dataset properties
-      auto* obstacle_properties = obstacle_outline->mutable_properties();
-      (*obstacle_properties)["borderWidth"] = "2";
-      (*obstacle_properties)["pointRadius"] = "0";
-      (*obstacle_properties)["lineTension"] = "0";
-      (*obstacle_properties)["fill"] = "false";
-      (*obstacle_properties)["showLine"] = "true";
+  int obstacle_index = 1;
+  for (const auto& obstacle : open_space_debug.obstacles()) {
+    auto* obstacle_outline = chart->add_line();
+    obstacle_outline->set_label("Bdr" + std::to_string(obstacle_index));
+    obstacle_index += 1;
+    for (int vertice_index = 0;
+         vertice_index < obstacle.vertices_x_coords_size(); vertice_index++) {
+      auto* point_debug = obstacle_outline->add_point();
+      point_debug->set_x(obstacle.vertices_x_coords(vertice_index));
+      point_debug->set_y(obstacle.vertices_y_coords(vertice_index));
     }
+    // Set chartJS's dataset properties
+    auto* obstacle_properties = obstacle_outline->mutable_properties();
+    (*obstacle_properties)["borderWidth"] = "2";
+    (*obstacle_properties)["pointRadius"] = "0";
+    (*obstacle_properties)["lineTension"] = "0";
+    (*obstacle_properties)["fill"] = "false";
+    (*obstacle_properties)["showLine"] = "true";
+  }
+
+  auto smoothed_trajectory = open_space_debug.smoothed_trajectory();
+  auto* smoothed_line = chart->add_line();
+  smoothed_line->set_label("Smooth");
+  size_t adc_label = 0;
+  for (const auto& point : smoothed_trajectory.vehicle_motion_point()) {
+    const auto& x = point.trajectory_point().path_point().x();
+    const auto& y = point.trajectory_point().path_point().y();
+    const auto& heading = point.trajectory_point().path_point().theta();
```

Comment:
For scalar types just store the value, which gives you best locality.

Reply:
@xiaoxq Done